docs/fundamentals/code-analysis/quality-rules/ca2255.md

```diff
@@ -0,0 +1,40 @@
+---
+title: "CA2255: The `ModuleInitializer` attribute should not be used in libraries"
+description: "Learn about code analysis rule CA2255: The `ModuleInitializer` attribute should not be used in libraries"
+ms.date: 11/15/2021
+ms.topic: reference
+f1_keywords:
+- CA2255
+- ModuleInitializerAttributeShouldNotBeUsedInLibraries
+helpviewer_keywords:
+- ModuleInitializerAttributeShouldNotBeUsedInLibraries
+- CA2255
+author: jeffhandley
+---
+# CA2255: The `ModuleInitializer` attribute should not be used in libraries
+
+| | Value |
+|-|-|
+| **Rule ID** |CA2255|
+| **Category** |[Usage](usage-warnings.md)|
+| **Fix is breaking or non-breaking** |Breaking|
+
+## Cause
+
+Applying <xref:System.Runtime.CompilerServices.ModuleInitializerAttribute> to a method within a Class Library.
+
+## Rule description
+
+Module initializers are intended to be used by application code to ensure an application's components are initialized before the application code begins executing. If library code declares a <xref:System.Runtime.CompilerServices.ModuleInitializerAttribute> method, it can interfere with application initialization and also lead to limitations in that application's trimming abilities. Library code should therefore not utilize the <xref:System.Runtime.CompilerServices.ModuleInitializerAttribute> attribute.
```

Comment:
```suggestion
Module initializers are intended to be used by application code to ensure an application's components are initialized before the application code begins executing. If library code declares a method with the <xref:System.Runtime.CompilerServices.ModuleInitializerAttribute>, it can interfere with application initialization and also lead to limitations in that application's trimming abilities. Library code should therefore not utilize the <xref:System.Runtime.CompilerServices.ModuleInitializerAttribute>.
```

Reply:
Probably should consider this feedback on the analyzer description in:

https://github.com/dotnet/roslyn-analyzers/blob/e2d60adee15089b72a7f91cf49f1207bb34dd8ca/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx#L1771

Reply:
That is so much better, @IEvangelist; thanks!